### PR TITLE
Fix unavailable gRPC match functions causing cycle to extend to `proposalCollectionInterval`

### PR DESCRIPTION
**Pull request by HazWard**
**What this PR does / Why we need it**:
When `FetchMatches` is called on gRPC endpoints that are not available the cycle gets extended to the `proposalCollectionInterval` instead of failing the function call immediately. These changes check if the function is available before making a `MatchFunctionClient.Run` call to get results. 

**Which issue(s) this PR fixes**:

Closes #

**Special notes for your reviewer**:

### CODE REVIEW

File: internal/app/backend/backend_service.go [hunk 1]
```diff
@@ -367,6 +373,21 @@ func doAssignTickets(ctx context.Context, req *pb.AssignTicketsRequest, store st
 	return resp, nil
 }
 
+func checkMmfAvailability(address string) error {
+	timeout := 1 * time.Second
```
Comment (HazWard) [inline internal/app/backend/backend_service.go]: The timeout should probably be configurable here.

**Review by cristian-sandru — APPROVED** [review]
LGTM

**Comment by guillaumed-unity** [comment]
That looks good for a P1 bug fix. 
Another option that I think of, a bit more advanced, would be to keep a list of available match functions that are being kept up to date through a background process and regularly refreshed. We would check against that list before doing a call.

**Review by guillaumed-unity — APPROVED** [review]
That looks good for a P1 bug fix.
Another option that I think of, a bit more advanced, would be to keep a list of available match functions that are being kept up to date through a background process and regularly refreshed. We would check against that list before doing a call.

**Review by cristian-sandru — APPROVED** [review]
LGTM after changes

**Pull request closed without merging**